pkg/manager/task_scan.go

```diff
@@ -150,19 +151,43 @@ func (t *ScanTask) scanScene() {
 	}
 }
 
-func (t *ScanTask) makeScreenshots(probeResult ffmpeg.VideoFile, checksum string) {
+func (t *ScanTask) makeScreenshots(probeResult *ffmpeg.VideoFile, checksum string) {
 	thumbPath := instance.Paths.Scene.GetThumbnailScreenshotPath(checksum)
 	normalPath := instance.Paths.Scene.GetScreenshotPath(checksum)
 
 	thumbExists, _ := utils.FileExists(thumbPath)
 	normalExists, _ := utils.FileExists(normalPath)
+
 	if thumbExists && normalExists {
 		logger.Debug("Screenshots already exist for this path... skipping")
 		return
 	}
 
-	t.makeScreenshot(probeResult, thumbPath, 5, 320)
-	t.makeScreenshot(probeResult, normalPath, 2, probeResult.Width)
+	if probeResult != nil {
+		// makescreenshots is called for a new file
+		t.makeScreenshot(*probeResult, thumbPath, 5, 320)
+		t.makeScreenshot(*probeResult, normalPath, 2, probeResult.Width)
+	} else {
+		// makescreenshots is called for an already existing file
+		if !thumbExists || !normalExists {
```

Comment:
Regarding probeResult if it's nil i don't necessary need to create it
The whole point in the ifs is to avoid as much as possible creating it even if  it means a little more code , because to create  it means running ffprobe for the media file 
When the file is new we can't avoid it because we need the probe anyway but when it's in the database me must avoid that as much as possible
Also any call to makeScreenshot must be spared because it calls ffmpeg to create an image from the media file

Reply:
I agree. That's why the `probeResult` creation would be after the check for `thumbExists` and `normalExists`. It would only be created if `probeResult` is nil and a screenshot needs to be created.

Reply:
lol now fixing 
realized what you meant